tests/engine/processes/workchains/test_restart.py

```diff
@@ -8,44 +8,94 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """Tests for `aiida.engine.processes.workchains.restart` module."""
-from aiida.backends.testbase import AiidaTestCase
-from aiida.engine.processes.workchains.restart import BaseRestartWorkChain
-from aiida.engine.processes.workchains.utils import process_handler
+# pylint: disable=invalid-name
+import pytest
 
+from aiida.engine import CalcJob, BaseRestartWorkChain, process_handler, ProcessState, ProcessHandlerReport, ExitCode
 
-class TestBaseRestartWorkChain(AiidaTestCase):
-    """Tests for the `BaseRestartWorkChain` class."""
 
-    @staticmethod
-    def test_is_process_handler():
-        """Test the `BaseRestartWorkChain.is_process_handler` class method."""
+class SomeWorkChain(BaseRestartWorkChain):
+    """Dummy class."""
 
-        class SomeWorkChain(BaseRestartWorkChain):
-            """Dummy class."""
+    _process_class = CalcJob
```

Comment:
What is this? I don't remember needing to set a `_process_class` property inside workchains...is this new?

Reply:
This is just a concept of the `BaseRestartWorkChain`. Note that this is an implementation of a `WorkChain` but not a concrete one, i.e., you cannot run a `BaseRestartWorkChain`. It is merely a baseclass for a very common type of `WorkChain`: one that runs a `CalcJob` and deals with errors. So to run an *actual* base restart workchain, one should subclass it and define this `_process_class` which will be the `CalcJob` that the workchain will launch. Look at the `BaseRestartWorkChain` docstring for more information